packages/core/src/code-loading/webRequire.js

```diff
@@ -110,6 +111,11 @@ const makeWebRequire = (filesAndFolders, options) => {
           _compile: (content, fileName) => {
             const moduleMakerFunction = new Function('require', 'module', content)
             moduleMakerFunction(_require.bind(null, entry.fullPath), matchingModule)
+
+            const paramDefFromSource = isMain ? getParameterDefinitionsFromSource(content, fileName) : []
```

Comment:
It's more important to parse definitions from the source WHEN there are definitions to be parsed, i.e. @jscad-params is present. That check only required a cheap scan of the content.